cpp/src/arrow/io/buffered_test.cc

```diff
@@ -293,6 +293,14 @@ TEST_F(TestBufferedOutputStream, Tell) {
 
   ASSERT_OK(buffered_->Close());
 
+  // write large buffer after raw_pos is cached
+  OpenBuffered(3, false);
+  AssertTell(0);
+  ASSERT_OK(buffered_->Write(std::string(10, '0').data(), 5));
```

Comment:
Looks to me this code is deferencing a dangling pointer. Please define a string variable.

Reply:
Oops. Thanks for catching.